devito/types.py

```diff
@@ -74,6 +74,7 @@ class Basic(object):
     is_SparseFunction = False
     is_PrecomputedSparseFunction = False
     is_PrecomputedSparseTimeFunction = False
+    is_TimeDifferentiable = False
```

Comment:
I don't like this. I actually preferred the way it was before.
What we should rather do (not necessarily in this PR, up to u) is introducing a mixin class for Time-depending Functions. Thus, both `TimeFunction` and `AbstractSparseTimeFunction` could inherit from it and also share lots of redundant code in `__init__` as well as the `time_order` property and other stuff

Reply:
for nest PR, reverted